### ISSUE TITLE
Backslash incorrectly removed.

### ISSUE THREAD

**Issue by mAarnos** [rust-lang-nursery/rustfmt#716]
The code 

```
println!("forall x. forall y. forall z. mult(x, mult(y, z)) = mult(mult(x, y), z) /\\
          forall x. mult(e(), x) = x /\\
          forall x. mult(x, x) = e()
          ==> forall x. forall y. mult(x, y) = mult(y, x)");
```

gets formatted to

```
println!("forall x. forall y. forall z. mult(x, mult(y, z)) = mult(mult(x, y), z) /\forall \
          x. mult(e(), x) = x /\forall x. mult(x, x) = e()
          ==> forall x. \
          forall y. mult(x, y) = mult(y, x)");
```

which is clearly incorrect as it is no longer possible to compile the code. Also, the formatting is absolutely horrible but that is a separate issue. This happens with the latest development version (compiled with 1.5 stable), and the version installed with 'cargo install rustfmt' (again, compiled with 1.5 stable). I am using Windows 7 but I don't believe that is relevant.

**Comment by nrc** [comment]
Other than fixing the backslash bug, I wonder if we could do better at preserving formatting in string literals - seems to be a common problem that people have significant spacing which rustfmt then stomps all over.

**Comment by nrc** [comment]
huh, this triggers an ICE.

I suspect once the ICE is fixed, this will be fixed by #726.

**Comment by nrc** [comment]
No ICE actually, that was something else.

It seems that we now insert some *extra* slashes:

```
if expando.ptr.is_null() {
        println!("forall x. forall y. forall z. mult(x, mult(y, z)) = mult(mult(x, y), z) /\\
              \
                  forall x. mult(e(), x) = x /\\
              forall x. mult(x, x) = e()
              \
                  ==> forall x. forall y. mult(x, y) = mult(y, x)");
    }
```

**Comment by drusellers** [comment]
> Other than fixing the backslash bug, I wonder if we could do better at preserving formatting in string literals - seems to be a common problem that people have significant spacing which rustfmt then stomps all over.

Is there an issue for this? `rustfmt` is mangling my inline SQL statements.

**Comment by nrc** [comment]
Well, we don't muck up the formatting any more, but we do still remove the slash, which is weird. Investigating...

**Comment by nrc** [comment]
No actually, that is a lie, we totally leave it alone

**Comment by davidalber** [comment]
Here's how it is formatted today (0.2.15-nightly) with default configuration:
```rust
println!(
    "forall x. forall y. forall z. mult(x, mult(y, z)) = mult(mult(x, y), z) /\\
          forall x. mult(e(), x) = x /\\
          forall x. mult(x, x) = e()
          ==> forall x. forall y. mult(x, y) = mult(y, x)"
);
```

Diff from original:
```diff
Diff in /Users/davidalber/dev/foo/716.rs at line 1:
-println!("forall x. forall y. forall z. mult(x, mult(y, z)) = mult(mult(x, y), z) /\\⏎
+println!(⏎
+    "forall x. forall y. forall z. mult(x, mult(y, z)) = mult(mult(x, y), z) /\\⏎
           forall x. mult(e(), x) = x /\\⏎
           forall x. mult(x, x) = e()⏎
-          ==> forall x. forall y. mult(x, y) = mult(y, x)");⏎
+          ==> forall x. forall y. mult(x, y) = mult(y, x)"⏎
+);⏎
 ⏎
```

If I put it inside a function the spacing turns out a little differently, but it otherwise the same.
```rust
fn main() {
    println!(
        "forall x. forall y. forall z. mult(x, mult(y, z)) = mult(mult(x, y), z) /\\
          forall x. mult(e(), x) = x /\\
          forall x. mult(x, x) = e()
          ==> forall x. forall y. mult(x, y) = mult(y, x)"
    );
}
```

@nrc: does the "only-with-option" label mean that this requires a non-default configuration value to repro a problem? If yes, what is it?

**Comment by nrc** [comment]
I assumed that the option to re-format string literals (`format_strings` and/or `force_format_strings`) was required since otherwise we should not touch string literals at all.

**Comment by davidalber** [comment]
I see. With some combinations of `format_strings` and `max_width` I can get extra backslashes. Here's a rustfmt.toml:
```toml
format_strings = true
max_width = 50
```

Then the example from my last comment formats to the following.
```rust
println!(
    "forall x. forall y. forall z. mult(x, \
     mult(y, z)) = mult(mult(x, y), z) /\\
          \
     forall x. mult(e(), x) = x /\\
          \
     forall x. mult(x, x) = e()
          ==> \
     forall x. forall y. mult(x, y) = mult(y, x)"
);
```

This seems like the issue also described in #1210.

Is the extra backslash the only remaining issue in here? My reading of the thread is that the original issue (removal of a backslash) is resolved. Have I got that right?

**Comment by nrc** [comment]
@davidalber  thanks for investigating this! Yes it seems like the original issue is resolved and what is left is a dup of #1210. So, lets close this as a dup.
